File: backend/internal/entity/item.go
package entity

// Item represents single in-game item.
type Item struct {
	Id           int    `json:"id"`
	Name         string `json:"name"`
	ImageSrc     string `json:"imageSrc"`
	ImageForHero string `json:"imageForHero"`
	Description  string `json:"description"`
	Price        int    `json:"price"`
	Category     string `json:"category"`
	Armor        int    `json:"armor"`
	Damage       int    `json:"damage"`
	Rarity       string `json:"rarity"`

	State ItemState `json:"state"`
}

// State represents state.
type State string

// ItemState represents item state.
type ItemState State

const (
	// Unknown shows that the ItemState is unknown for server.
	Unknown ItemState = ""
	// Equipped shows that user is using this item.
	Equipped ItemState = "equipped"
	// Inventoried shows that item is in inventory.
	Inventoried ItemState = "inventoried"
	// Store shows that item in a store
	Store ItemState = "store"
)

File: frontend/js/inventory/item.ts
type Item = {
    id: number
    name: string
    imageSrc: string
    description: string
    price: number
    category: Category
    rarity: Rarity
    state: ItemState
}

type Rarity = 'common' | 'rare' | 'epic' | 'legendary'

function getRarityColor(rarity: Rarity): string {
    switch (rarity) {
        case 'common':
            return '#C8C8C8'
        case 'rare':
            return '#0b9ccf'
        case 'epic':
            return '#d461f7'
        case 'legendary':
            return '#ff9c00'
        default:
            return 'linear-gradient(#40E0D0, #91e047, #fff456, #fff456, #ffa856, #e64f4f)'
    }
}

type ItemState = 'store' | 'equipped' | 'inventoried'
type Category = 'helmet' | 'chest' | 'leggins' | 'boots' | 'weapon' | 'pet'
